Buildscripts/DevicetreeCompiler/source/generator.py

```
@@ -57,44 +57,69 @@ def find_phandle(devices: list[Device], phandle: str):
+        if device_property.name not in binding_property_names:
+            raise DevicetreeException(f"Device '{device.node_name}' has invalid property '{device_property.name}'")
+
     # Allocate total expected configuration arguments
     result = [0] * len(binding_properties)
     for index, binding_property in enumerate(binding_properties):
         device_property = find_device_property(device, binding_property.name)
         if device_property is None:
-            if binding_property.required:
-                raise Exception(f"device {device.node_name} doesn't have property '{binding_property.name}'")
+            if binding_property.default is not None:
+                temp_prop = DeviceProperty(
```

Ruff (0.14.14)

[error] `DeviceProperty` may be undefined, or defined from star imports

(F405)